src/service/payout-request-pdf-service.ts

```diff
@@ -0,0 +1,108 @@
+import {PdfGenerator} from "../entity/file/pdf-file";
+import PayoutRequestPdf from "../entity/file/payout-request-pdf";
+
+const PDF_GEN_URL =  process.env.PDF_GEN_URL ? process.env.PDF_GEN_URL : 'http://localhost:3001/pdf';
+
+export default class PayoutRequestPdfService {
+
+  static pdfGenerator: PdfGenerator = {
+    client: new Client(PDF_GEN_URL, { fetch }),
+    fileService: new FileService(PAYOUT_REQUEST_PDF_LOCATION),
+  };
+
+
+  /**
+   * Constructs and returns the parameters required for generating a payout request PDF.
+   * @param payoutRequest - The payout request for which to generate the parameters.
+   * @returns {PayoutParameters} An instance of `PayoutParameters` containing all necessary information for PDF generation.
+   */
+  static getParameters(payoutRequest: PayoutRequest): PayoutParameters {
+    return new PayoutParameters({
+      payout: new Payout({
+        bankAccountName: payoutRequest.bankAccountName,
+        bankAccountNumber: payoutRequest.bankAccountNumber,
+        amount: payoutRequest.amount.getAmount(),
+        reference: `SDS-PR-${payoutRequest.id}`,
+        date: payoutRequest.createdAt,
+        debtorNumber: String(payoutRequest.requestedBy.id),
+      }),
+    });
+  }
+
+  /**
+   * Prepares and returns the parameters required by the PDF generator client to create a payout request PDF.
+   * @param payoutRequest - The payout request for which to generate the parameters.
+   * @returns {PayoutRouteParams} An instance of `PayoutRouteParams` containing the consolidated parameters and settings for PDF generation.
+   */
+  static getPdfParams(payoutRequest: PayoutRequest): PayoutRouteParams {
+    const params = this.getParameters(payoutRequest);
+
+    const settings: FileSettings = new FileSettings({
+      createdAt: new Date(),
+      fileType: ReturnFileType.PDF,
+      language: Language.ENGLISH,
+      name: '',
+      stationery: 'BAC',
+    });
+
+    const data: IPayoutRouteParams = {
+      params,
+      settings,
+    };
+
+    return new PayoutRouteParams(data);
+  }
+
+  /**
+   * Generates a PDF for a payout request and uploads it to the file service.
+   * If the payout request PDF generation or upload fails, it throws an error with the failure reason.
+   * @param payoutRequestId - The ID of the payout request to generate and upload the PDF for.
+   * @returns {Promise<PayoutRequestPdf>} A promise that resolves to the `PayoutRequestPdf` entity representing the generated and uploaded PDF.
+   */
+  public static async createPdf(payoutRequestId: number): Promise<PayoutRequestPdf> {
+    const payoutRequest = await PayoutRequest.findOne({
+      where: { id: payoutRequestId },
+      relations: ['requestedBy', 'approvedBy', 'payoutRequestStatus'],
+    });
+    if (!payoutRequest) return undefined;
+
+    const params = this.getPdfParams(payoutRequest);
+    return this.pdfGenerator.client.generatePayout(params).then(async (res: FileResponse) => {
```

Comment:
Please use a try catch block here instead. In https://github.com/GEWIS/sudosos-backend/pull/205/commits/a477c0801f134c253c941ee0cf1ca27ac00e0ff1, I also changed it, because due to incorrect parameters the `.generatePayout()` function did not return a promise but an empty object. This results in an error that you cannot call .then on undefined, which is a really, really weird error. Awaiting the generation and putting everything in a try-catch block will result in better errors.